### PR TITLE
Milestone #1: Create Book List that Allows for Adding and Removing Books

### PR DESCRIPTION
**Pull request by Anjacodes**
In this milestone, we: 

* Created an HTML skeleton
* Created add and remove functions
* Dynamically added books entered by the user to the top of the page 
* Stored added books in local storage and made sure they get retrieved from local storage on page load.

### CODE REVIEW

**Comment by omar-labana** [comment]
Hi team, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project where the linter checks are not passing.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: myscripts.js [hunk 1]
```diff
@@ -1,26 +1,77 @@
...
-addbook = function() {
-    const titleinput = document.getElementById("titleInput").value;
-    const authorinput = document.getElementById("authorInput").value;
+};
+
+// OBJECT CONSTRUCTOR FUNCTION
+
+window.addEventListener('load', () => {
+  storageObject = JSON.parse(localStorage.getItem('books'));
+  createBook(storageObject);
```
Comment (henatan99) [inline myscripts.js]: - [ ] While the page loads initially the local storage is empty, therefore the `storageObject` has undefined or may be a null value. As a result, the `createBook(storageObject)` call will throw an error. Kindly fix this issue. :pray: 

Screenshot from 2021-11-24 01-57-53

File: myscripts.js [hunk 2]
```diff
@@ -1,26 +1,77 @@
...
-    
+  for (let i = 0; i < elements.length; i += 1) {
+    elements[i].addEventListener('click', () => {
+      // console.log('remove clicked');
+      if (bookCont[i].parentNode) {
+        bookCont[i].parentNode.removeChild(bookCont[i]);
+      }
+      collection.splice(i, 1);
+      localStorage.setItem('books', JSON.stringify(collection));
+    });
```
Comment (henatan99) [inline myscripts.js]: - [ ] I think this block of code is buggy. When I remove a book from the list in the UI, it removes all the book objects from the local storage. Please carefully hunt the bug and fix the issue. :pray: 

Screenshot from 2021-11-24 02-00-54

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @Anjacodes , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: myscripts.js (hunk 1 above)
Comment (Anjacodes) [reply myscripts.js]: Hi @henatan99,
Thank you for bringing this to our attention!
Somehow, we didn't get that error but implemented a change that should take care of the issue.

File: myscripts.js (hunk 2 above)
Comment (Anjacodes) [reply myscripts.js]: You are so right @henatan99, thank you for pointing this out. This error is now fixed.

File: myscripts.js [hunk 3]
```diff
@@ -1,26 +1,74 @@
...
+};
+
+const update = () => {
+  const elements = document.getElementsByClassName('remove');
+  const bookCont = document.querySelectorAll('.bookCont');
+  collection = JSON.parse(localStorage.getItem('books'));
+  Array.from(elements).forEach((btn, index) => {
+    btn.addEventListener('click', () => {
+      if (bookCont[index].parentNode) {
+        bookCont[index].parentNode.removeChild(bookCont[index]);
```
Comment (omar-labana) [inline myscripts.js:33]: - [ ] I tested the app and when I create two or more book and remove them then refresh the page one of the removed book is still present in the `localStorage`. Kindly investigate this issue.

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi team, 

Excellent work so far! 👌
You have implemented almost all the required features. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: myscripts.js (hunk 3 above)
Comment (Anjacodes) [reply myscripts.js:33]: Hi @omar-labana , 

Thank you for the feedback. We have fixed this error as you suggested.

**Review by AshakaE — APPROVED** [review]
### Hi @Anjacodes   🙋‍♀️, 

# STATUS APPROVED ✔
   
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**